slrealizer/realize_sdss.py

```diff
@@ -0,0 +1,125 @@
+            raise ValueError("Need to define either lensID or rownum, not both.")
+        
+        if lensID is not None:
+            return self.catalog.get_lens(lensID) # Don't use this (?)
+        elif rownum is not None:
+            return self.catalog.sample[rownum]
+       
+    def _from_sdss_to_galsim(self, lensInfo, band):
+        """
+        Converts SDSS column values into GalSim terms
+        
+        Keyword arguments:
+        lensInfo -- a row of the OM10 DB
+        band -- the filter used to observe
+
+        Returns:
+        A dictionary (named galsimInput) containing properties that 
+        can be passed into GalSim (See code for which properties)
+        """
+        
+        # We will input flux in units of nMgy
+        mag   = lensInfo[band + '_SDSS_lens'] 
+        flux  = from_mag_to_flux(mag, to_unit='nMgy') # in nMgy
+        hlr   = lensInfo['REFF_T'] # REFF_T is in arcsec
+        e     = lensInfo['ELLIP']
+        beta  = lensInfo['PHIE'] * galsim.degrees # PHIE is in degrees
+        
+        galsimInput={'flux': flux,
+                       'half_light_radius': hlr,
+                       'e': e,
+                       'beta': beta,
+                       'num_objects': lensInfo['NIMG']}
+        
+        for obj in xrange(lensInfo['NIMG']):
+            obj_mag = lensInfo[band + '_SDSS_quasar']\
+                      + from_flux_to_mag(abs(lensInfo['MAG'][obj]))
+                      #+ from_flux_to_mag(lensInfo['MAG'][obj] + get_filter_AB_offset())
+            galsimInput['flux_'+str(obj)] = from_mag_to_flux(obj_mag, to_unit='nMgy') # don't subtract AB offset?
+            galsimInput['xy_'+str(obj)] = lensInfo['XIMG'][obj], lensInfo['YIMG'][obj]
+            
+        return galsimInput        
+            
+    def draw_system(self, obsInfo, lensInfo, save_dir=None):
+        galsimInput = self._from_om10_to_galsim(lensInfo, obsInfo['filter'])
+        return self.as_super.draw_system(galsimInput=galsimInput, obsInfo=obsInfo, save_dir=save_dir)
+    
+    def estimate_hsm(self, obsInfo, lensInfo):
+        galsim_img = self.draw_system(lensInfo=lensInfo, obsInfo=obsInfo, save_dir=None)
+        return self.as_super.estimate_hsm(galsim_img=galsim_img, obsInfo=obsInfo)
+    
+    def draw_emulated_system(self, obsInfo, lensInfo):
+        hsmOutput = self.estimate_hsm(obsInfo, lensInfo)
+        return self.as_super.draw_emulated_system(hsmOutput)
+    
+    def create_source_row(self, obsInfo, lensInfo):
+        hsmOutput = self.estimate_hsm(obsInfo, lensInfo)
+        if hsmOutput is None:
+            return None
+        objectId = lensInfo['LENSID']
+        return self.as_super.create_source_row(hsmOutput=hsmOutput, objectId=objectId, obsInfo=obsInfo)
+    
+    def make_source_table(self, save_dir):
+        return self.as_super.make_source_table(save_dir=save_dir)
+    
+    def draw_lens_random_date(self, lensID=None, rownum=None, save_dir=None):
```

Comment:
This looks very similar to the other subclass's method - maybe the super class could handle most of these operations?

Reply:
Moved to parent class and made `draw_emulated_image` method replace the task of drawing the emulated image.